logstash-core/src/main/java/org/logstash/instrument/witness/PluginsWitness.java

```diff
@@ -50,12 +64,23 @@ public PluginWitness filters(String id) {
     }
 
     /**
+     * Gets the {@link PluginWitness} for the given id, creates the associated {@link PluginWitness} if needed
+     *
+     * @param id the id of the codec
+     * @return the associated {@link PluginWitness} (for method chaining)
+     */
+    public PluginWitness codecs(String id) {
```

Comment:
whoops, this snuck into the review. 

Turns out that we record codec metrics as a plugin (shares the same Ruby code with the Input).  I am not sure if that desired behavior or accidental. 

I haven't run across any consumption of this, but didn't want to change the Ruby code, so I opted to simply support codecs as new plugin metrics.